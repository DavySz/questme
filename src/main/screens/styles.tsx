import { GestureHandlerRootView } from "react-native-gesture-handler";
import styled from "styled-components/native";

export const GestureWrapper = styled(GestureHandlerRootView)`
  flex: 1;
`;
