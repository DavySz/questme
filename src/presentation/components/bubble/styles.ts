import styled, { css } from "styled-components/native";
import { ICircleStyles, IPositionWrapperStyles } from "./types";

export const getWrapperPosition = (
  position: Partial<IPositionWrapperStyles>
) => {
  const customPosition = [];

  if (position.bottom) {
    customPosition.push(css`
      bottom: ${position.bottom}px;
    `);
  }

  if (position.left) {
    customPosition.push(css`
      left: ${position.left}px;
    `);
  }

  if (position.top) {
    customPosition.push(css`
      top: ${position.top}px;
    `);
  }

  if (position.right) {
    customPosition.push(css`
      right: ${position.right}px;
    `);
  }

  return customPosition;
};

export const CircleFill = styled.View<ICircleStyles>`
  width: ${({ size }) => size}px;
  height: ${({ size }) => size}px;

  border-radius: ${({ size }) => size / 2}px;

  background-color: ${({ theme }) => theme.colors.support["accent-2"]};

  opacity: 0.1;
`;

export const CircleOutline = styled.View<ICircleStyles>`
  width: ${({ size }) => size}px;
  height: ${({ size }) => size}px;

  border-radius: ${({ size }) => size / 2}px;

  background-color: ${({ theme }) => theme.colors.transparent};
  border: 1px solid ${({ theme }) => theme.colors.support["accent-2"]};

  opacity: 0.1;
`;

export const PositionWrapper = styled.View<Partial<IPositionWrapperStyles>>`
  position: absolute;
  ${(props) => getWrapperPosition(props as Partial<IPositionWrapperStyles>)};
`;
