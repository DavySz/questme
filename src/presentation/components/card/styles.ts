import styled from "styled-components/native";

export const Wrapper = styled.View`
  flex-direction: row;

  align-items: center;
  justify-content: space-between;

  width: 100%;
  padding: 8px;

  border-radius: 20px;
  border: 2px solid ${({ theme }) => theme.colors.neutral["grey-5"]};

  background-color: ${({ theme }) => theme.colors.neutral.white};
`;

export const Content = styled.View`
  gap: 6px;
`;

export const Group = styled.View`
  gap: 6px;
`;
