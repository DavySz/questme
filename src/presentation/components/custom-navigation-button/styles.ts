import styled from "styled-components/native";

export const Wrapper = styled.TouchableOpacity`
  border-radius: 24px;
  height: 48px;
  width: 48px;

  justify-content: center;
  align-items: center;

  margin-top: -20px;
`;

export const Container = styled.View`
  border-radius: 32px;
  height: 64px;
  width: 64px;

  justify-content: center;
  align-items: center;

  background-color: ${({ theme }) => theme.colors.transparent};

  margin-top: -20px;
`;
