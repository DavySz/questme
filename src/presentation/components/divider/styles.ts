import styled from "styled-components/native";

export const Line = styled.View`
  width: 100%;
  height: 1px;
  flex-direction: row;
  background-color: ${({ theme }) => theme.colors.neutral["gey-4"]};
  margin: 0 16px;
`;
