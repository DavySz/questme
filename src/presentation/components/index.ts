export * from "./button";
export * from "./text";
export * from "./screen-template";
export * from "./divider";
