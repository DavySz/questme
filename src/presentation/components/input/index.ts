import { InputIcon } from "./icon";
import { InputLabel } from "./label";
import { InputRoot } from "./root";
import { InputText } from "./text";

export const Input = {
  Root: InputRoot,
  Label: InputLabel,
  Text: InputText,
  Icon: InputIcon,
};
