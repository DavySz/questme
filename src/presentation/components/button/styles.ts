import styled, { css } from "styled-components/native";
import {
  IButtonIconStyles,
  IRootStyle,
  TButtonVariant,
  TIconSide,
} from "./types";
import { TTheme, TThemeColors } from "@presentation/styles";

export const getButtonBackground = (
  variant: TButtonVariant,
  theme: TTheme
): string => {
  const colors: Record<TButtonVariant, string> = {
    primary: theme.colors.primary,
    secondary: theme.colors.neutral["grey-5"],
    tertiary: theme.colors.transparent,
    facebook: theme.colors.company.facebook,
    google: theme.colors.neutral.white,
    location: theme.colors.neutral.white,
  };

  return colors[variant];
};

export const getButtonTextColor = (variant: TButtonVariant): TThemeColors => {
  const color: Record<TButtonVariant, TThemeColors> = {
    primary: "neutral-white",
    secondary: "primary",
    tertiary: "primary",
    facebook: "neutral-white",
    google: "neutral-black",
    location: "primary",
  };

  return color[variant] as TThemeColors;
};

const isFull = (isFull: boolean) => {
  if (isFull) {
    return css`
      width: 100%;
    `;
  }
};

const getIconSide = (side: TIconSide) => {
  if (side === "left") {
    return css`
      margin-right: 16px;
    `;
  }

  if (side === "right") {
    return css`
      margin-left: 16px;
    `;
  }

  return null;
};

const getBorder = (variant: TButtonVariant, theme: TTheme) => {
  if (variant === "google") {
    return css`
      border: 1px solid ${theme.colors.neutral["grey-4"]};
    `;
  }
};

export const Button = styled.TouchableOpacity.attrs({
  activeOpacity: 0.7,
})<IRootStyle>`
  ${({ full }) => isFull(full)};
  ${({ variant, theme }) => getBorder(variant, theme as TTheme)};

  background-color: ${({ theme, variant }) =>
    getButtonBackground(variant, theme as TTheme)};

  border-radius: 20px;
  padding: 16px;

  flex-direction: row;

  align-items: center;
  justify-content: center;
`;

export const IconWrapper = styled.View<IButtonIconStyles>`
  ${({ side }) => getIconSide(side)}
`;
