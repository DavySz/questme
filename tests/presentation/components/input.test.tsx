import { Input } from "@presentation/components/input";
import { theme } from "@presentation/styles";
import { act, render, waitFor } from "@testing-library/react-native";
import { ThemeProvider } from "styled-components/native";

import LockIcon from "@presentation/assets/icons/lock-icon.svg";
import { View } from "react-native";
describe("Input", () => {
  test("Should render input without errors", () => {
    const tree = render(
      <ThemeProvider theme={theme}>
        <Input.Label>test-label</Input.Label>
        <Input.Root>
          <Input.Icon icon={LockIcon} />
          <Input.Text />
        </Input.Root>
      </ThemeProvider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("Should update isFocused state to true when onFocus has been called", async () => {
    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <View>
          <View testID="view-outside" />
          <Input.Label>test-label</Input.Label>
          <Input.Root>
            <Input.Icon icon={LockIcon} />
            <Input.Text />
          </Input.Root>
        </View>
      </ThemeProvider>
    );

    const inputWrapper = getByTestId("input-wrapper");

    act(() => {
      getByTestId("custom-input").props.onFocus();
    });

    await waitFor(() => expect(inputWrapper.props.isFocused).toBe(true));
  });

  test("Should update isFocused state to false when onBlur has been called", async () => {
    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <View>
          <View testID="view-outside" />
          <Input.Label>test-label</Input.Label>
          <Input.Root>
            <Input.Icon icon={LockIcon} />
            <Input.Text />
          </Input.Root>
        </View>
      </ThemeProvider>
    );

    const inputWrapper = getByTestId("input-wrapper");

    expect(inputWrapper.props.isFocused).toBe(false);

    act(() => {
      getByTestId("custom-input").props.onFocus();
    });

    await waitFor(() => expect(inputWrapper.props.isFocused).toBe(true));

    act(() => {
      getByTestId("custom-input").props.onBlur();
    });

    await waitFor(() => expect(inputWrapper.props.isFocused).toBe(false));
  });
});
