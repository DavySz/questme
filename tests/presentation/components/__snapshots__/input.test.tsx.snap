// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Should render input without errors 1`] = `
[
  <View
    style={
      {
        "marginBottom": 8,
      }
    }
  >
    <Text
      color="neutral-black"
      style={
        {
          "color": "#0C092A",
          "fontFamily": "Rubik_400Regular",
          "fontSize": 14,
          "lineHeight": 20,
        }
      }
      variant="body-small-regular"
    >
      test-label
    </Text>
  </View>,
  <View
    isFocused={false}
    style={
      {
        "backgroundColor": "#FFF",
        "borderBottomLeftRadius": 20,
        "borderBottomRightRadius": 20,
        "borderColor": "#EFEEFC",
        "borderStyle": "solid",
        "borderTopLeftRadius": 20,
        "borderTopRightRadius": 20,
        "borderWidth": 2,
        "flexDirection": "row",
        "gap": 16,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
        "width": "100%",
      }
    }
    testID="input-wrapper"
  >
    <SvgMock />
    <TextInput
      onBlur={[Function]}
      onFocus={[Function]}
      style={
        {
          "fontFamily": "Rubik_400Regular",
          "fontSize": 16,
          "width": "100%",
        }
      }
      testID="custom-input"
    />
  </View>,
]
`;
